sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/JreCertificates.java

```diff
@@ -0,0 +1,181 @@
+        if (null != JRE_KEY_STORE) {
+            try {
+                jreKsAliases = Collections.unmodifiableList(Collections.list(JRE_KEY_STORE.aliases()));
+            } catch (KeyStoreException e) {
+                LOGGER.log(WARNING, "Unable to load the jre key store aliases.", e);
+            }
+        }
+        ALIASES = jreKsAliases;
+        CERTS = ALIASES.stream()
+            .collect(Collectors.toMap(a -> a, a -> {
+                try {
+                    return JRE_KEY_STORE.getCertificate(a);
+                } catch (KeyStoreException e) {
+                    LOGGER.log(WARNING, "Unable to get the jre key store certificate.", e);
+                }
+                return null;
+            }));
+    }
+
+    /**
+     * Private constructor
+     */
+    private JreCertificates() {
+
+    }
+
+    /**
+     *
+     * @return the singleton.
+     */
+    public static JreCertificates getInstance() {
+        return INSTANCE;
+    }
+
+
+    @Override
+    public List<String> getAliases() {
+        return ALIASES;
+    }
+
+    @Override
+    public Map<String, Certificate> getCertificates() {
+        return CERTS;
+    }
+
+    @Override
+    public Map<String, Key> getCertificateKeys() {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public void deleteEntry(String alias) {
+
+    }
+
+    private static class JREKeyStore {
+        private static final String JAVA_HOME =  privilegedGetProperty("java.home", "");
+        private static final Path STORE_PATH = Paths.get(JAVA_HOME).resolve("lib").resolve("security");
+        private static final Path DEFAULT_STORE = STORE_PATH.resolve("cacerts");
+        private static final Path JSSE_DEFAULT_STORE = STORE_PATH.resolve("jssecacerts");
+        private static final String KEY_STORE_PASSWORD = privilegedGetProperty("javax.net.ssl.keyStorePassword", "changeit");
+
+        private static KeyStore getDefault() {
+            KeyStore defaultKeyStore = null;
+            try {
+                defaultKeyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+                loadKeyStore(defaultKeyStore);
+            } catch (KeyStoreException e) {
+                LOGGER.log(WARNING, "Unable to get the jre key store.", e);
+            }
+            return defaultKeyStore;
+        }
+
+        private static void loadKeyStore(KeyStore ks) {
+            InputStream inStream = null;
+            try {
+                inStream = Files.newInputStream(getKeyStoreFile());
+                ks.load(inStream, KEY_STORE_PASSWORD.toCharArray());
+            } catch (IOException | NoSuchAlgorithmException | CertificateException e) {
+                LOGGER.log(WARNING, "unable to load the jre key store", e);
+            } finally {
+                try {
+                    inStream.close();
+                } catch (NullPointerException | IOException e) {
+                    LOGGER.log(WARNING, "", e);
+                }
+            }
+        }
```

Comment:
Can we do like this:
```
        private static void loadKeyStore(KeyStore ks) {
            try (InputStream inStream = Files.newInputStream(getKeyStoreFile())) {
                ks.load(inStream, KEY_STORE_PASSWORD.toCharArray());
            } catch (IOException | NoSuchAlgorithmException | CertificateException e) {
                LOGGER.log(WARNING, "unable to load the jre key store", e);
            }
        }
```

Reply:
checkstyle will complain with failures. " redundant non-null check."